Add note on broken react runtime types (#2383)
mdx-js/mdx: docs/migrating/v3.mdx

If you passed the `useDynamicImport` option before, remove it, the behavior
is now the default.

> Note that you may get a TypeScript error saying `Property 'Fragment' is missing in type`.
> This is [because `react/jsx-runtime` does not export its types](https://github.com/orgs/mdx-js/discussions/1862#discussioncomment-1824570).
> To remediate this, do:
> 
> ```typescript
> 
> // @ts-expect-error: the automatic react runtime is untyped.
> const result = await run(compiled, { ...runtime, baseUrl: import.meta.url });
> ```

Reviewer: ````suggestion
If you use `react/jsx-runtime`, you might get a TypeScript error (such as
`Property 'Fragment' is missing in type`), because it is typed incorrectly.
To remediate this, do:

```tsx
import {type Fragment, type Jsx, run} from '@mdx-js/mdx'
import * as runtime_ from 'react/jsx-runtime'

// @ts-expect-error: the automatic react runtime is untyped.
const runtime: {Fragment: Fragment; jsx: Jsx; jsxs: Jsx} = runtime_

const result = await run('# hi', {...runtime, baseUrl: import.meta.url})
```
````
